continuum/rehearsal/memory.py

```diff
@@ -0,0 +1,132 @@
+from typing import Union, Callable, Tuple, Any
+
+import numpy as np
+
+from continuum.rehearsal import herd_random, herd_closest_to_cluster, herd_closest_to_barycenter
+
+
+class RehearsalMemory:
```

Comment:
I think this class should inherit from task set because they are quite similar. RehearsalMemory class should only add some specific features to a taskset class, IMO.

Reply:
I though about it. If RehearsalMemory we can indeed wrap a loader around only rehearsal memory, although we usually merge this memory with the main dataset.

But the problem is that we would need to provide the transformations to the rehearsal memory which is a bit clumsy I think. And most people will probably simply merge it with the main dataset as I show in the doc.

So I don't agree, but we can add a method `get_taskset` which takes transformations as argument and return a TaskSet view of the memory.

Reply:
So, the rehearsalMemory should be compatible with Transformation Scenarios? For "image_path" scenario, do we save the path or the image? 
Maybe for "complex" scenario we can create a memory composed of a set of sub-taskset, so we can keep track easily of transformations

Reply:
In the order:
- no
- by default path, but that can be easily changed by the user
- I think it's safer to first consider scenario where the transformations is the same for all tasks. After all 99% of the papers are about that